### PR TITLE
Create Readme.md file for having release notes

### PR DESCRIPTION
**Pull request by eldhosembabu**
For Dynamic links iOS SDK, we don't have a place to communicate the latest changes in the release to 3p developers. 

Adding a readme.md file to inform developers on the recent changes in SDK release and other info that developers should be aware of.

### CODE REVIEW

File: FirebaseDynamicLinks/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Firebase Dynamic Links are links that work the way you want, on multiple platforms, and whether or not your app is already installed.
+For more info : https://firebase.google.com/products/dynamic-links
+
+Please visit [our developer site]
+(https://firebase.google.com/docs/dynamic-links/) for integration instructions,
+documentations, support information, and terms of service.
+
+## Managing the Pasteboard
+
+### V 4.2.0
```
Comment (paulb777) [inline FirebaseDynamicLinks/README.md:12]: Remove this line

File: FirebaseDynamicLinks/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+This release of the FDL SDK adds a plist property (FirebaseDeepLinkPasteboardRetrievalEnabled) that a developer can set to enable/disable the use of iOS pasteboard by the SDK.
```
Comment (paulb777) [inline FirebaseDynamicLinks/README.md:14]: s/This release of the FDL SDK adds/Use/

File: FirebaseDynamicLinks/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+FDL SDK uses the pasteboard for deep-linking post app install (to enable deferred deep-linking, where the link is copied on the [app preview page](https://firebase.google.com/docs/dynamic-links/link-previews#app_preview_pages)) and app install attribution;  otherwise, FDL does not use the pasteboard for anything else. 
+
+Disabling pasteboard access affects the app in the following ways:
+Deferred deep-linking will not work as reliably.  At best, your app receives [weak matches](https://firebase.google.com/docs/reference/unity/namespace/firebase/dynamic-links#linkmatchstrength) for deep-links.
+App install attribution stats will be less accurate (potentially undercounting app installs).
+
+Enabling pasteboard access affects the app in the following ways:
+On iOS 14, will show a system alert notifying that your app accessed the content in the pasteboard. This should happen one-time after installation of the app.
```
Comment (paulb777) [inline FirebaseDynamicLinks/README.md:23]: Use markdown bullets: Prefix each line with ` *`

File: FirebaseDynamicLinks/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Deferred deep-linking will work as designed.  At best, your app receives a [perfect match](https://firebase.google.com/docs/reference/unity/namespace/firebase/dynamic-links#linkmatchstrength) for deep-links.
+SDK will be able to more reliably attribute installation stats for links
```
Comment (paulb777) [inline FirebaseDynamicLinks/README.md:25]: Missing period

File: FirebaseDynamicLinks/README.md (hunk 1 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: Done

File: FirebaseDynamicLinks/README.md (hunk 2 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: done

File: FirebaseDynamicLinks/README.md (hunk 3 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: Done

File: FirebaseDynamicLinks/README.md (hunk 4 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: Done

**Comment by paulb777** [comment]
Please fix the trailing whitespace causing the GHA failures

File: FirebaseDynamicLinks/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,8 @@
 - [fixed] Reduce frequency of iOS14 pasteboard notifications by only reading from it when it contains URL(s). (#5905)
 - [changed] Functionally neutral updated import references for dependencies. (#5824)
 
+Refer to [README.md](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDynamicLinks/README.md) for more details about this release.
```
Comment (morganchen12) [inline FirebaseDynamicLinks/CHANGELOG.md:7]: nit: This should be "Refer to the README" instead of "Refer to README"

File: FirebaseDynamicLinks/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Firebase Dynamic Links SDK for iOS
+
+Firebase Dynamic Links are links that work the way you want, on multiple platforms, and whether or not your app is already installed.
```
Comment (morganchen12) [inline FirebaseDynamicLinks/README.md:3]: Rephrase this to be more clear and less market-y.

"Firebase Dynamic Links are universal deep links that persist across app installs."

File: FirebaseDynamicLinks/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+For more info : https://firebase.google.com/products/dynamic-links
```
Comment (morganchen12) [inline FirebaseDynamicLinks/README.md:4]: `For more info, see the [Firebase website](https://firebase.google.com/products/dynamic-links).`

File: FirebaseDynamicLinks/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+Please visit [our developer site]
+(https://firebase.google.com/docs/dynamic-links/) for integration instructions,
+documentations, support information, and terms of service.
+
+## Managing the Pasteboard
+
+This release of the FDL SDK use a plist property (FirebaseDeepLinkPasteboardRetrievalEnabled) that a developer can set to enable/disable the use of iOS pasteboard by the SDK.
```
Comment (morganchen12) [inline FirebaseDynamicLinks/README.md:12]: nit: replace `use` with `adds`

File: FirebaseDynamicLinks/CHANGELOG.md (hunk 5 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/CHANGELOG.md]: added "the"

File: FirebaseDynamicLinks/README.md (hunk 8 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md:12]: it was changed from adds to use based on an early comment. Please confirm if this change back is must.

File: FirebaseDynamicLinks/README.md (hunk 7 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: done

File: FirebaseDynamicLinks/README.md (hunk 8 above)
Comment (morganchen12) [reply FirebaseDynamicLinks/README.md:11]: Hm. Maybe leave it as `use` and replace `This release of the FDL SDK` with `Firebase Dynamic Links 4.2.0 and higher`?

I'm also ok with just ignoring my nit comment above.

Comment (ryanwilson) [reply FirebaseDynamicLinks/README.md]: +1 for replacing the "This release" with "Version 4.2.0 and higher"

Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md]: Done

File: FirebaseDynamicLinks/README.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Firebase Dynamic Links SDK for iOS
```
Comment (ryanwilson) [inline FirebaseDynamicLinks/README.md:1]: Nit: can you please format this to ensure the lines don't go over 100 characters? For links you can see how we do it in the regular README:

https://raw.githubusercontent.com/firebase/firebase-ios-sdk/master/README.md

Comment (eldhosembabu) [reply FirebaseDynamicLinks/README.md:1]: I did the formating where ever possible. please see
